### ISSUE TITLE
Shaman - Feral Spirit ability

### ISSUE THREAD

**Issue by Lucika77** [chromiecraft/chromiecraft#2593]
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

### Current Behaviour

When you summon wolves from the ability Feral Spirit, they are always on agressive mode and have Bash + Leap on auto. 

### Expected Blizzlike Behaviour

I think you should be able to turn off/on their abilities + their mode (agressive/defensive/passive) and when you resummon the wolves, they should keep these settings. For example: You summon the wolves, turn Bash off, you resummon the wolves, the Bash should be off.

### Steps to reproduce the problem

Get the ability Feral Spirit, summon the wolves
They are on agressive mode and Bash + Leap on auto
Turn off the Bash ability + put them on passive and resummon the wolves
It stays the same - agressive mode, Bash + Leap on auto

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/d87d0a2ae892286a44a85cc230fa66d44479d4bd

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by Shineslippers** [comment]
Thank you for the report. For now this behavior can considered as bllizlike. I have been searching for more info, but except some videos from wrath times there is not enought sources. Our dev said ` Temporary pets should not save its pet bars.` https://github.com/azerothcore/azerothcore-wotlk/issues/7263 Anyway there is something wrong.

**Comment by Plasmashield** [comment]
That is, the video where the shaman calls the wolves and the autocast is turned off is not enough? That is, warlocks and magicians should be the same?

**Comment by Shineslippers** [comment]
No, autocast is everywhere is enabled. There is question about which spells should be affected by this all or not. Also question about stances. Not everywhere is `agressive`. And yes mages, priests and etc has same behavior of temporary pets.

**Comment by Plasmashield** [comment]
Yes, I heard that temporary pets do not save bars changes
but
https://www.warcraftmovies.com/movieview.php?id=154835

**Comment by Shineslippers** [comment]
Cool why you didn't show it to UltraNix when you had chance.

**Comment by Salehyn** [comment]
Tested by Bench on WOTLK classic. Wolves are set to default autocasts but if you turn it off, the settings will remember the next time you cast the spell. So this report is valid.

Waiting for confirmation on other temporary pets.

**Comment by laasker** [comment]
> Tested by Bench on WOTLK classic. Wolves are set to default autocasts but if you turn it off, the settings will remember the next time you cast the spell. So this report is valid.
> 
> Waiting for confirmation on other temporary pets.

I don't have a mage on classic to test the Water Elemental, but i was able to test priest Shadowfiend and druid Force of Nature and both works like the Wolves that you described

sorry for the potato quality idk what's happening with my OBS
Shadowfiend:
https://streamable.com/qdl1w5
https://streamable.com/jsk75x
on the second video you can see that it saves the passive state and the autocast is off on Shadowcrawl

Druid treants:
https://streamable.com/il8kjy
https://streamable.com/0zmt8a
It doesnt have any ability, but it saves the passive state

**Comment by heyitsbench** [comment]
Could be that being in combat affects whether the preferences are saved across summons, still needs some deeper testing.

**Comment by heyitsbench** [comment]
Figured out that there is absolutely something wrong with AC/CC's implementation (with hard proof!). The SummonProperties flag Save Pet Autocast is set for priest's Shadowfiend, mage's Water Elemental, druid's Force of Nature, as well as shaman's Feral Spirits. This flag is entirely unimplemented on AC, need to figure out proper implementation. I've mentioned this before, but preferences are saved between summons, but sometimes get forgotten/overwritten, it's very strange.

**Comment by FrancescoBorzi** [comment]
> Figured out that there is absolutely something wrong with AC/CC's implementation (with hard proof!)

@heyitsbench can you please link this to AC then?

**Comment by paper43-hash** [comment]
This is still a bug?
